app/models/testcase_run.rb

```diff
@@ -1,4 +1,11 @@
 class TestcaseRun < ApplicationRecord
   belongs_to :testcase
   belongs_to :build
+
+  validates_presence_of :testcase_id
+  validates_presence_of :time
+  validates_presence_of :passed
+  validates_presence_of :skipped
+  validates_presence_of :build_id
```

Comment:
we can also set build and testcase to non-null in the db. might simplify the model